pkg/logql/step_evaluator.go

```diff
@@ -4,10 +4,22 @@ import (
 	"github.com/prometheus/prometheus/promql"
 )
 
+type StepResult interface {
+	PromVec() promql.Vector
+}
+
+type PromVec promql.Vector
```

Comment:
I'm wondering if we should move away from the Prometheus terminology... can we perhaps just call this `Vector` or `Samples`?

Reply:
Hm, vector is too generic as we'll have other vector implementations but we could call it `SampleVector`.